### ISSUE TITLE
Linting broken for script tag (auto-fix mangles file)

### ISSUE THREAD

**Issue by falco467** [ota-meshi/eslint-plugin-astro#190]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.36.0

### What version of `eslint-plugin-astro` are you using?

0.26.0

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
  extends: [
    'plugin:astro/recommended',
    'standard-with-typescript'
  ],
  parserOptions: {
    project: './tsconfig.json'
  }
}

```
</details>

```astro
<!DOCTYPE html>
<html lang="de">
  <head>
    <script>
      alert('test')
    </script>
  </head>
  <body>
  </body>
</html>
```

### What did you expect to happen?

I expected no errors and --fix to correctly format the file

### What actually happened?

ESLint shows errors:
```
  6:4  error  Trailing spaces not allowed                                no-trailing-spaces
  6:4  error  Too many blank lines at the end of file. Max of 0 allowed  no-multiple-empty-lines
  6:5  error  Newline required at end of file but not found              eol-last
```

And when running `eslint --fix` on the file, the result is completely broken:
```
<!DOCTYPE html>
<html lang="de">
  <head>
    <script>
      alert('test')
   <!DOCTYPE html>
<html lang="de">
  <head>
    <script>
      alert('test')

<!DOCTYPE html>
<html lang="de">
  <head>
    <script>
      alert('test')
    </script>
  </head>
  <body>
  </body>
</html>
```

### Link to Minimal Reproducible Example

https://ota-meshi.github.io/eslint-plugin-astro/playground/#eJyVWcuuHLcR/ZWbu3ECmBCcpaFk42QdL7wJoA2nm9NDq5tskex52PC/51Tx3dMjORAEDauq+ajHqUPq9/fBjur9x/ePf/nXf3765b8///vtEpb5n5/MR/r3bZZm+sen91F9eofs7e3jRcmRf+G3H5xeQxq9vclZufDX74Ly4bu/JZMPjc3HD+Xjjyc7PpIw/YQ2Lvz+/bvbZuXff/z9Xfrg7AdjxWDNedZDEF4FMWqnhqCvZPP376vRqFYoZFCjYJkYpLFGD3Le3PwN07MKwwWrBGXCN0yd8na+qt6K9kUHqJt71gd1D6/0m9k8Vhi8h+kMC+sODUZ11kaJq3ReaCN8eMztTNjqWTkxzBITzdofr5es7OlXqBrjZxO/zvqFhVeLboZXOWvs3y6r7nb0q78L+cMPjw9yjh441JnhYp2Qy0lPm938twwv0ovnYO2ttBe8q0MTp6WQ7JhR+UGZUZrA8wZ50mZU95dfrc6u/qva8FjVawtnXziIlJvx27paR+mGPFhwwuOZtmDJ2bMK6vUpB9TMZ6GuNAsOd1Xis3qk8aE9fIrtsR8QcjtozvtZntR8ZE8Frc30rXhwYZANwcmRgT47uajofh2O3aOXCaU3bjFSyKZDI+zBxbCK5EgvznbYDo/L59ofNvng2P54+4saETp2glyDtubQCHmt/o9YoOblMCgUHuxeGVDKXPQ4KiOsiQeVp2P/kTmS5qUzCOpQ4ew9hPRryQfT1tHJVAQrCG8bzatUN0+WeY4is+bV0i++DP2m/vzSXyt4Y5ucoylfbcoHGfTwp4+R4DXISdgrfrzaLck5sZz6sgHJR/F1CGL7kvhft/WDXQ8XzQ6Bo8RqvS7dA6UBy42aEy2iYos6Y5S6TMr8SYVAh1Jhc1FCyecfZqDdLJrq4K6GLTc5avASwlGh0zhZvUYK7j34q6cyFQGfdEoYNYnflLNVbsYnU2z5SEYBJXtddh01jH9OTSkfmACctmlSro4T6FYJfAFnT74TxJ0vajkp12vUjMPqcydDkbdGAG6gEbslS9WCxtINxHCRlGW54/fKVVIUytkQz94NaLFOipNFykhDK4VeUxo8JGc5z+Hi7DZdimhDOPsZp9me5LwT6oWScS80kS+wo9cidRhvSAFxu2jwyFUO5fSzhTPtWRAZ076JGdJpTm3ohTsWfUfh8GzIJKQIErwojboJ/1hOCfEjPNgh4BxIR73gX/ADuZaNgDXBWfNcZmDjPEB+B0v9X5w2PQNJixmKl9NbVQHOnraVZeB+552r/FEx+Ysc7Y2oaHCgxTieQQOt06A8PKWkkzWrwgWE6KRQrqqpXqaWYJV1gIAonnLZcIAZdLPqnJJwcu4wLPLyrARIKVzy2ElRn7ItryS23CPhWEylAcVT1TPH5bbsd0JivEVEFBnpcJLDZwZSZWqiZCXqZ7jshX0obzpEi4Su4qHVHIkU9WrtjYx7j8lKYbXRUexZTgPawx7pblJTC0DO2pLbBXSArdKh8d+Ryd734FPQtZ+w4kEspuIIbQxS4wg2Ka2N5P6GfG6mbhK1A+J9glEiJpwtCQRcmRmVcJ96xOuH68K32FGfNYUQJ3+G1yZ/Ot+kEK9OX3n6I+CkgKTkjXegLnAy4DSD2NYRE6Rux+QJB1ulTtMgaigaunA2F6fTbBFBboZjAfUB1TQX8EXOaNS4/o3p4VJ5Ud5puNjR8xapyFKvJHJ+1wmyKQJgV8QNGkc30vohOZd/4ZwSJVibQCtAlSTAzlKESi6N2AaEPtQCVF/oT2zZhN2EGGKhKskxLGJfh9VRE+CffHTg2GmTbhREBHRcS9NN1TzKlRHjWZkplRtGS6lOIEB4TuHZTnRxT1jInAoBgoVNSy7yHhMFmA4FEChtkxS4s6elaEQQ5vtR/ATnK94hFRR8A0hlXb9h19Yhkb2GHReczMnROI3qEOlToIHeSMoggshzbZ50uOna+Gk/Fa3pKWTz1O44nVv0sBWXiclos5XxqK89q2EO0td8pA2dV4o08VsulqL5Isw2l9anrrUNgkHgSp0QaEc4wC17Sb1eEq2YreQLSOq+DY0ABtJjhFVuaPZXFJF9+E5Ot+hmW9q0YWpgNAF8rkGShZhthX/wXbEhfX2Xmi3mZe90IoLJcxUAFMm9hZhIJDl82AFdZCt9spOYkmxHDaIM2VNRf8p32B55U2YTH0TzaZpE2y/ardE4vg+1kpuT69rsNfKkvqlGGALQd1td583T344nVSZUaAzaYtvgGs0uvI1m1xQbDd19UHBaHSlxHwnyXhVUCjsHZyE19NIZ+UVT5CdF7pVfNmIglX4xO6tts3M60egbUIjyq0zBFEwQFHYSJOvYChzaFDVAaaaWghkFbvkU3dRVK9F44k8ELE8y9K2N3JNfHTqdAQs5kO4BLKtAzNBNnqUt6jQldbW6HPcmnWFO39ZqerH0F8T7IhOOWFMvY+l3208Y6guLzXXVNBaWpwt5ZAi5AbQaPmUrGFU89VYoUFJQChvMq+PyecXWhLwy8mPRhnByY+3U26KQpDlJfKtLLqA3AXszBxqPoRxbRaZ6TrGBwhldN2e8iRytRuXSdr/yLLxfI3NDln3ZLGhcfXhwuJ7de75WKipLNqPpvyLaG6Gne2Nb3CwoF2UelWsB36MKPYsirvT4k294FDOu3oyM/IyOGIXERR92lPwjtuYTbpOfVSDoKwnUa2jVHP6oyS9Au2+QVUSl82sQC9qPEwttJVhCNUyCTibb0o+C9pMkaT+JtZxw8NFZE0HEqg1BhIcKfWTyR4XQfZPJQdRIN21Phy0mfN5ONSmTqg90wnUTly4eXUOfnJBjnxNFHHNM6ZGK0ys6kp5Q22kwvlk3drKOmcX3rHTgskrjs0gPJUrS9NkUFScVbgrHer6jMLdUlUxWktgjUGWLHV6rLkGS1/hLviUnWtoTp/aDwga69wSW4R6SWFxlwtSImlcQzEZ7mnaf12eYfPfKmZQsTMrbfNZ4ueqcnP+Xie+MbTZ0ijZgSVGStvso43efIascqQM2BIwkdCB2Y47ajrg3icRAF2Gt20t5BKMfT5ombzovNfvo5H1tNHpt2mBm2VnfRcbQKNv47aDIbhq3KGQpqGa3t91N/VBFb8+tIgPwyS48Jq4ntD6rOqoXCH4o2VfxH3/8Dzamc9Q=
